src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java

```diff
@@ -58,7 +58,18 @@ void testValidateChangesInvalidDate() {
         when(changesMock.getReleaseDate()).thenReturn(Optional.of(LocalDate.of(2020, 8, 1)));
         when(changesMock.getBody()).thenReturn(Optional.of("## Features"));
         when(changesMock.getFileName()).thenReturn("file");
-        assertThrows(IllegalStateException.class, () -> this.validator.validateChanges(changesMock, "2.1.0"));
+        assertThrows(IllegalStateException.class, () -> this.validator.validateChanges(changesMock, "2.1.0", true));
+    }
+
+    @Test
+    // [utest->dsn~validate-changes-file-contains-release-date~1]
+    void testValidateChangesInvalidDateWarning() {
+        final ReleaseLetter changesMock = Mockito.mock(ReleaseLetter.class);
+        when(changesMock.getVersionNumber()).thenReturn(Optional.of("2.1.0"));
+        when(changesMock.getReleaseDate()).thenReturn(Optional.of(LocalDate.of(2020, 8, 1)));
+        when(changesMock.getBody()).thenReturn(Optional.of("## Features"));
+        when(changesMock.getFileName()).thenReturn("file");
+        assertDoesNotThrow(() -> this.validator.validateChanges(changesMock, "2.1.0", false));
```

Comment:
Validate warning
You could inject a Mock logger using reflection and then use Mockito.verfiy to check that the method was called.

Reply:
I think there is no need to validate the logger message here. And I don't like to use reflection until it's 100% necessary

Reply:
ok